backend/src/routes/api/connection-types/connectionTypeUtils.ts

```diff
@@ -0,0 +1,191 @@
+import { V1ConfigMap } from '@kubernetes/client-node';
+import { KubeFastifyInstance } from '../../../types';
+import { getNamespaces } from '../../../utils/notebookUtils';
+
+const ODH_DASHBOARD_KEY = 'opendatahub.io/dashboard';
+const CONNECTION_TYPE_KEY = 'opendatahub.io/connection-type';
+
+const CONNECTION_TYPE_LABEL_SELECTOR = `${ODH_DASHBOARD_KEY} = true, ${CONNECTION_TYPE_KEY} = true`;
+
+const isConnectionTypeConfigMap = (configMap: V1ConfigMap): boolean => {
+  const labels = configMap.metadata.labels;
+  return labels[ODH_DASHBOARD_KEY] === 'true' || labels[CONNECTION_TYPE_KEY] === 'true';
```

Comment:
Can't labels be undefined? Also we need both labels to be present.
```suggestion
  return labels?.[ODH_DASHBOARD_KEY] === 'true' && labels?.[CONNECTION_TYPE_KEY] === 'true';
```